pkg/project/registry/project/proxy/proxy.go

```diff
@@ -167,6 +182,34 @@ func (s *REST) Update(ctx kapi.Context, name string, objInfo rest.UpdatedObjectI
 	return projectutil.ConvertNamespace(namespace), false, nil
 }
 
+func (s *REST) tryImpersonatingNamespaceUpdate(project *api.Project, ctx kapi.Context) (*kapi.Namespace, bool) {
+	user, ok := kapi.UserFrom(ctx)
+	if !ok {
+		return nil, false
+	}
+
+	impersonatingClient, err := authclient.NewImpersonatingKubernetesClientset(user, s.config)
```

Comment:
From an audit standpoint, this will look weird. Normally, you'd get:
1. request to update project
2. request to update namespace

Now you're getting:
1. request to update project
2. request to update project as someone else

This might look fishy in the logs and for someone doing security audit.